src/ensembl/production/metadata/api/genome.py

```diff
@@ -80,64 +112,118 @@ def fetch_taxonomy_ids(self, taxonomy_names):
         group_type = check_parameter(group_type)
 
+        # Construct the initial database query
         genome_select = db.select(
-            Genome, Organism, Assembly
-        ).join(Genome.assembly).join(Genome.organism)
-
+            Genome, Organism, Assembly, OrganismGroupMember, OrganismGroup
+        ).select_from(Genome) \
+            .join(Organism, Organism.organism_id == Genome.organism_id) \
+            .join(Assembly, Assembly.assembly_id == Genome.assembly_id) \
+            .join(OrganismGroupMember, OrganismGroupMember.organism_id == Organism.organism_id) \
+            .join(OrganismGroup, OrganismGroup.organism_group_id == OrganismGroupMember.organism_group_id) \
+
+        # Apply group filtering if group parameter is provided
         if group:
             group_type = group_type if group_type else ['Division']
-            genome_select = db.select(
-                Genome, Organism, Assembly, OrganismGroup
-            ).join(Genome.assembly).join(Genome.organism) \
-                .join(Organism.organism_group_members) \
-                .join(OrganismGroupMember.organism_group) \
-                .filter(OrganismGroup.type.in_(group_type)).filter(OrganismGroup.name.in_(group))
+            genome_select = genome_select.filter(OrganismGroup.type.in_(group_type)).filter(OrganismGroup.name.in_(group))
 
+        # Apply additional filters based on the provided parameters
         if unreleased_only:
+            # Question (Bilal): When adding unreleased genomes do we populate genome_releases table?
+            # if that's the case the test data doesn't do so (this will return en empty list)
             genome_select = genome_select.outerjoin(Genome.genome_releases).filter(
                 GenomeRelease.genome_id == None
             )
-        elif site_name is not None:
-            genome_select = genome_select.join(
-                Genome.genome_releases).join(
-                GenomeRelease.ensembl_release).join(
-                EnsemblRelease.ensembl_site).filter(EnsemblSite.name == site_name)
+        if is_released:
+            # Include release related info only if is_released is True
+            genome_select = genome_select.add_columns(EnsemblRelease, EnsemblSite) \
+                .join(GenomeRelease, Genome.genome_id == GenomeRelease.genome_id) \
+                .join(EnsemblRelease, GenomeRelease.release_id == EnsemblRelease.release_id) \
+                .join(EnsemblSite, EnsemblSite.site_id == EnsemblRelease.site_id)
+
+        if site_name is not None:
+            genome_select = genome_select.filter(EnsemblSite.name == site_name)
 
             if release_type is not None:
                 genome_select = genome_select.filter(EnsemblRelease.release_type == release_type)
 
             if current_only:
                 genome_select = genome_select.filter(GenomeRelease.is_current == 1)
 
-            if release_version is not None:
+            if release_version != 0.0:
```

Comment:
So when we are releasaing a genome we will create the genome_release and link the genome_datasets. Until then there is nothing for those genomes.
The 0.0 here is not proper for any aspect of the API

Reply:
done